crates/package_manager/src/add.rs

```diff
@@ -0,0 +1,96 @@
+use crate::Install;
+use derive_more::{Display, Error};
+use miette::Diagnostic;
+use pacquet_lockfile::Lockfile;
+use pacquet_npmrc::Npmrc;
+use pacquet_package_json::PackageJsonError;
+use pacquet_package_json::{DependencyGroup, PackageJson};
+use pacquet_registry::{PackageTag, PackageVersion};
+use pacquet_tarball::Cache;
+use reqwest::Client;
+
+/// This subroutine does everything `pacquet add` is supposed to do.
+#[must_use]
+pub struct Add<'a, ListDependencyGroups, DependencyGroupList>
+where
+    ListDependencyGroups: Fn() -> DependencyGroupList,
+    DependencyGroupList: IntoIterator<Item = DependencyGroup>,
+{
+    /// Shared cache that store downloaded tarballs.
+    pub tarball_cache: &'a Cache,
+    /// HTTP client to make HTTP requests.
```

Comment:
Is it really required to comment every single field? Most of the fields are self-explanatory. If they are not, then probably they should be renamed.

Reply:
For symmetry. I don't want to create an illusion where the doccomment for `tarball_cache` is for the group of `tarball_cache` + `http_client`.

Reply:
these comments have been removed by https://github.com/pnpm/pacquet/pull/143/commits/7ba09dfc34fec3f1da525abf9a5e7fd79d413f32